src/main.rs

```diff
@@ -325,57 +292,113 @@ async fn attest(
         not_after: Time::try_from(end).or(Err(ISE))?,
     };
 
-    // Create the basic subject alt name.
-    let mut sans = vec![GeneralName::DnsName(
-        Ia5StringRef::new("foo.bar.hub.profian.com").or(Err(ISE))?,
-    )];
+    let crs = Vec::<CertReq<'_>>::from_der(body.as_ref()).or(Err(StatusCode::BAD_REQUEST))?;
 
-    // Optionally, add the configured subject alt name.
-    if let Some(name) = state.san.as_ref() {
-        sans.push(GeneralName::DnsName(Ia5StringRef::new(name).or(Err(ISE))?));
-    }
+    // Decode and verify the certification request.
+    crs.into_iter()
+        .map(|cr| {
```

Comment:
This closure is too large to be a closure. You should split this into a separate function.

Reply:
It was done that way to address the lifetime issues, since `Certificate` requires a specific lifetime. I can re-do it using the `Vec<u8>` approach we discussed earlier. Roman had helped me to get this closure created.

Reply:
A function with defined lifetimes should work just as well. You shouldn't need to do the `Vec<u8>` approach.